### PR TITLE
Issues/adds post id to analytics

### PR DESCRIPTION
**Pull request by aerych**
Updates some of the reader analytics. 
- Removes some unnecessary nil checks.
- Adds post_id property to relevant analytics.
- Tracks feed_id and feed_item_id in lieu of blog_id and post_id for relevant analytics. 
- Adds URL property to analytics observing follows by URL 

cc @nbradbury and @blowery for property sanity check. Let's make sure we're using the same ones across platforms

@astralbodies would you be up for the code review, but please don't merge yet. 

Needs review: @astralbodies

### CODE REVIEW

**Comment by aerych** [comment]
Should be good to merge after review now. 
@astralbodies at your leisure sir.

**Comment by astralbodies** [comment]
@aerych There is a PR out there for logging Blog ID - #4540 by @daniloercoli. I'm just thinking through all the scenarios.

**Comment by aerych** [comment]
Looks like #4540 is also amending some of the issues addressed in this PR.  

Let's leave this one open for now, and wait til #4540 is merged since it has the larger scope. I'll make another pass after that.  Thanks @astralbodies.

**Comment by aerych** [comment]
@astralbodies Updated and ready for review :)

File: MIGRATIONS.md [hunk 1]
```diff
@@ -3,7 +3,12 @@
 This file documents changes in the data model. Please explain any changes to the
 data model as well as any custom migrations.
 
-## WordPress 43 (@koke 2015-11-23)
+## WordPress 43 (@aerych 2015-12-07
```
Comment (astralbodies) [inline MIGRATIONS.md]: Stupid but its missing the closing `)`

File: WordPress/Classes/Services/ReaderPostService.m [hunk 2]
```diff
@@ -137,10 +137,14 @@ - (void)toggleLikedForPost:(ReaderPost *)post success:(void (^)())success failur
         // Define success block. Make sure work is performed on the main queue
         void (^successBlock)() = ^void() {
             dispatch_async(dispatch_get_main_queue(), ^{
+                NSDictionary *properties = @{
+                                              WPAppAnalyticsKeyPostID: readerPost.postID,
+                                              WPAppAnalyticsKeyBlogID: readerPost.siteID
+                                              };
                 if (like) {
-                    [WPAppAnalytics track:WPAnalyticsStatReaderArticleLiked withBlogID:readerPost.siteID];
+                    [WPAnalytics track:WPAnalyticsStatReaderArticleLiked withProperties:properties];
```
Comment (astralbodies) [inline WordPress/Classes/Services/ReaderPostService.m]: Actually can you switch this to use the Track method on `WPAppAnalytics` - @daniloercoli added convenience methods that pass through the event tracking to WPAnalytics. Another layer but it at least makes all of the tracking calls go through `WPAppAnalytics` app-wide.

File: WordPress/Classes/Services/ReaderPostService.m [hunk 3]
```diff
@@ -137,10 +137,14 @@ - (void)toggleLikedForPost:(ReaderPost *)post success:(void (^)())success failur
...
                 } else {
-                    [WPAppAnalytics track:WPAnalyticsStatReaderArticleUnliked withBlogID:readerPost.siteID];
+                    [WPAnalytics track:WPAnalyticsStatReaderArticleUnliked withProperties:properties];
```
Comment (astralbodies) [inline WordPress/Classes/Services/ReaderPostService.m]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/Services/ReaderSiteService.m [hunk 4]
```diff
@@ -149,7 +147,8 @@ - (void)followSiteAtURL:(NSString *)siteURL success:(void(^)())success failure:(
             if (success) {
                 success();
             }
-            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed];
+            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed withProperties:@{ @"URL":sanitizedURL }];
```
Comment (astralbodies) [inline WordPress/Classes/Services/ReaderSiteService.m]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/Services/ReaderSiteService.m [hunk 5]
```diff
@@ -173,7 +172,7 @@ - (void)unfollowSiteAtURL:(NSString *)siteURL success:(void(^)())success failure
         if (success) {
             success();
         }
-        [WPAnalytics track:WPAnalyticsStatReaderSiteUnfollowed];
+        [WPAnalytics track:WPAnalyticsStatReaderSiteUnfollowed withProperties:@{@"URL":siteURL}];
```
Comment (astralbodies) [inline WordPress/Classes/Services/ReaderSiteService.m]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 6]
```diff
@@ -782,10 +782,18 @@ - (void)tapRecognized:(id)sender
...
-        
+        NSMutableDictionary *properties = [NSMutableDictionary dictionary];
+        properties[WPAppAnalyticsKeyBlogID] = post.siteID;
+        properties[WPAppAnalyticsKeyPostID] = post.postID;
+        properties[WPAppAnalyticsKeyIsJetpack] = @(post.isJetpack);
+        if (post.feedID && post.feedItemID) {
+            properties[WPAppAnalyticsKeyFeedID] = post.feedID;
+            properties[WPAppAnalyticsKeyFeedItemID] = post.feedItemID;
+        }
+        [WPAnalytics track:WPAnalyticsStatReaderArticleCommentedOn withProperties:properties];
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/ViewRelated/Reader/ReaderPostDetailViewController.m [hunk 7]
```diff
@@ -431,11 +431,17 @@ - (void)bumpStats
...
-    
-    [WPAppAnalytics track:WPAnalyticsStatReaderArticleOpened withProperties:properties withBlogID:self.post.siteID];
+    properties[WPAppAnalyticsKeyPostID] = self.post.postID;
+    properties[WPAppAnalyticsKeyBlogID] = self.post.siteID;
+    properties[WPAppAnalyticsKeyIsJetpack] = @(self.post.isJetpack);
+    if (self.post.feedID && self.post.feedItemID) {
+        properties[WPAppAnalyticsKeyFeedID] = self.post.feedID;
+        properties[WPAppAnalyticsKeyFeedItemID] = self.post.feedItemID;
+    }
+    [WPAnalytics track:WPAnalyticsStatReaderArticleOpened withProperties:properties];
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostDetailViewController.m]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 8]
```diff
@@ -468,7 +468,7 @@ import WordPressComAnalytics
         tableView.setContentOffset(CGPoint.zero, animated: true)
     }
 
-    private func propertyForStats() -> [NSObject: AnyObject] {
+    private func topicPropertyForStats() -> [NSObject: AnyObject] {
         assert(readerTopic != nil, "A reader topic is required")
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: I know this isn't new code but should we use NSAssert so an exception is throw rather than the app just crashing? nitpick I know :stuck_out_tongue:

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 9]
```diff
@@ -939,7 +952,7 @@ import WordPressComAnalytics
             }
         }
 
-        WPAnalytics.track(.ReaderInfiniteScroll, withProperties: propertyForStats())
+        WPAnalytics.track(.ReaderInfiniteScroll, withProperties: topicPropertyForStats())
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 10]
```diff
@@ -1260,7 +1273,7 @@ import WordPressComAnalytics
         let controller = ReaderStreamViewController.controllerWithSiteID(post.siteID, isFeed: post.isExternal)
         navigationController?.pushViewController(controller, animated: true)
 
-        let properties = NSDictionary(object: post.blogURL, forKey: "URL") as! [NSObject : AnyObject]
+        let properties = statsPropertiesForPost(post, andValue: post.blogURL, forKey: "URL")
         WPAnalytics.track(.ReaderSitePreviewed, withProperties: properties)
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: s/WPAnalytics/WPAppAnalytics

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 11]
```diff
@@ -1282,7 +1295,7 @@ import WordPressComAnalytics
         let controller = ReaderStreamViewController.controllerWithTagSlug(post.primaryTagSlug)
         navigationController?.pushViewController(controller, animated: true)
 
-        let properties = NSDictionary(object: post.primaryTagSlug, forKey: "tag") as! [NSObject : AnyObject]
+        let properties =  statsPropertiesForPost(post, andValue: post.primaryTagSlug, forKey: "tag")
         WPAnalytics.track(.ReaderTagPreviewed, withProperties: properties)
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: s/WPAnalytics/WPAppAnalytics

**Comment by astralbodies** [comment]
Once those WPAnalytics -> WPAppAnalytics changes are made, :shipit:

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 8 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Nice thought. Let's do it in a different pr tho.

**Comment by aerych** [comment]
Changes made.   Thanks @astralbodies!
